app/models/list.model.js

```
@@ -0,0 +1,69 @@
+        ListData.createAttachmentList = function (title, attachment) {
+            if (typeof title === 'string') {
+                var array = [];
+                for (var index = 0; (attachment instanceof Array) && (index < attachment.length); index++) {
+                    var element = attachment[index];
+                    var item = AttachmentData.create(element.attachmentTitle, element.link, element.publicity, element.buttonType);
+                    if (item) {
+                        array.push(item);
+                    }
+                }
+                return new ListData(title, array);
+            }
+            return null;
+        };
+
+        ListData.createDecisionList = function (title, decision) {
+            if (typeof title === 'string') {
+                var array = [];
+                for (var index = 0; (decision instanceof Array) && (index < decision.length); index++) {
+                    var element = decision[index];
+                    var item = AttachmentData.create(element.decisionTitle, element.link, element.publicity, element.buttonType);
+                    if (item) {
+                        array.push(item);
+                    }
+                }
+                return new ListData(title, array);
+            }
+            return null;
+        };
+
+        ListData.createAdditionalMaterialList = function (title, material) {
+            if (typeof title === 'string') {
+                var array = [];
+                for (var index = 0; (material instanceof Array) && (index < material.length); index++) {
+                    var element = material[index];
+                    var item = AttachmentData.create(element.additionalMaterialTitle, element.link, element.publicity, element.buttonType);
+                    if (item) {
+                        array.push(item);
+                    }
+                }
+                return new ListData(title, array);
+            }
+            return null;
```

Add an error log print for bad args cases, otherwise they're silently ignored.